lib/include/kframework/avm/teal/teal-execution.md

```diff
@@ -227,43 +227,55 @@ teal, failure means undoing changes made to the state (for more details, see
   rule <k> #calcReturn() => .K ... </k>
        <stack> I : .TStack </stack>
        <stacksize> SIZE </stacksize>
+       <paniccode> 0 </paniccode>
        <returncode> 4 => 0 </returncode>
        <returnstatus> _ => "Success - positive-valued singleton stack" </returnstatus>
     requires I >Int 0 andBool SIZE ==Int 1
 
   rule <k> #calcReturn() => .K ... </k>
        <stack> I : .TStack </stack>
        <stacksize> _ </stacksize>
+       <paniccode> 0 </paniccode>
        <returncode> 4 => 1 </returncode>
        <returnstatus> _ => "Failure - zero-valued singleton stack" </returnstatus>
     requires 0 >=Int I
 
   rule <k> #calcReturn() => .K ... </k>
        <stack> _ </stack>
        <stacksize> SIZE </stacksize>
+       <paniccode> 0 </paniccode>
        <returncode> 4 => 2 </returncode>
        <returnstatus> _ => "Failure - stack size greater than 1" </returnstatus>
     requires SIZE >Int 1
 
   rule <k> #calcReturn() => .K ... </k>
        <stack> .TStack </stack>
+       <paniccode> 0 </paniccode>
        <returncode> 4 => 2 </returncode>
        <returnstatus> _ => "Failure - empty stack" </returnstatus>
 
   rule <k> #calcReturn() => .K ... </k>
        <stack> (_:Bytes) : .TStack </stack>
        <stacksize> _ </stacksize>
+       <paniccode> 0 </paniccode>
        <returncode> 4 => 2 </returncode>
        <returnstatus> _ => "Failure - singleton stack with byte array type" </returnstatus>
 
+  rule <k> #calcReturn() => .K ... </k>
+       <paniccode> PANIC_CODE </paniccode>
+       <panicstatus> S </panicstatus>
+       <returncode> 4 => 3 </returncode>
+       <returnstatus> _ => "Failure - panic: " +String S </returnstatus>
+    requires PANIC_CODE =/=Int 0
+
   // Leave the testing commands on the K cell
   rule <k> #stopIfError() ~> X:TestingCommand => X:TestingCommand ~> #stopIfError() ... </k>
 
   // Consume the rest of the K cell if the execution terminated with an error
-  rule <k> #stopIfError() ~> (_:KItem => .K) ... </k>
+  rule <k> #stopIfError() ~> (ITEM:KItem => .K) ... </k>
        <returncode> RETURN_CODE </returncode>
     requires RETURN_CODE =/=Int 0
-
+     andBool notBool(isTestingCommand(ITEM))
```

Comment:
I don't think this condition is actually required. Let's add it for now though to be safe. We can test with `--coverage` later to see if it ever gets triggered.